server/test/integration/serverTest.js

```diff
@@ -60,13 +63,33 @@ describe('Brave Sensor server', () => {
       })
 
       it('should log the error', () => {
-        expect(helpers.logError).to.be.calledWithExactly('Bad request to /api/sensorEvent: coreid (Invalid value),event (Invalid value)')
+        expect(helpers.logError).to.be.calledWithExactly(
+          'Bad request to /api/sensorEvent: coreid (Invalid value),event (Invalid value),api_key (Invalid value)',
+        )
+      })
+    })
+
+    describe('for an otherwise valid DURATION request with an incorrect API key', () => {
+      beforeEach(async () => {
+        await firmwareAlert(chai, server, radar_coreID, SENSOR_EVENT.DURATION, badpassword)
+      })
+
+      it('should return 200', () => {
+        expect(this.response).to.have.status(200)
+      })
+
+      it('should log the error', () => {
+        expect(helpers.logError).to.be.calledWithExactly(`Access not allowed: incorrect key ${badpassword}`)
+      })
+
+      it('should not start the alert session since access should not have been given', () => {
+        expect(braveAlerter.startAlertSession).to.not.be.calledOnce
```

Comment:
What do you think of using `to.not.be.called` instead of `to.not.be.calledOnce`. Having the "Once" in there makes me wonder whether it would pass if it were called more than once.

Reply:
oh interesting, thank you for pointing that out! i'll change it and test it again!